### PR TITLE
Admin can create, edit,delete subject and subject's task

### PR DESCRIPTION
**Pull request by tunv**

### CODE REVIEW

File: app/controllers/admin/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      redirect_to [:admin, @subject]
+    else
+      render :edit
+    end
+  end
+
+  private
+  def subject_params
+    params.require(:subject).permit :name, :description,
+    tasks_attributes: [:id, :name, :description]
```
Comment (leminhtuan2015) [inline app/controllers/admin/subjects_controller.rb]: Convention

File: app/views/admin/subjects/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  <%= form_for [:admin, @subject] do |f| %>
+    <%= render "shared/error_messages", object: f.object %>
+
+    <%= f.label :name %>
+    <%= f.text_field :name, class: "form-control" %>
+
+    <%= f.label :description %>
+    <%= f.text_field :description, class: "form-control" %>
+    
+    <h3><%= t "Admin.task"%></h3>
```
Comment (leminhtuan2015) [inline app/views/admin/subjects/_form.html.erb]: Space

File: app/views/admin/subjects/_task_fields.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<div class="form-group" style="border:0px solid black; overflow:hidden">
+  <%= f.label :Task %>
+  <%= f.text_field :name, calss: "form-control" %>
+  <%= f.label :Description %>
+  <%= f.text_field :description, class: "form-control" %>
+  <div class= "btn btn-info", style="float: right;">
```
Comment (leminhtuan2015) [inline app/views/admin/subjects/_task_fields.html.erb]: It is better move style to css

File: app/views/admin/subjects/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<% provide :title, t("Admin.All_subject") %>
+<h1><%= t "Admin.All_subject" %></h1>
+<ul class = "users">
```
Comment (leminhtuan2015) [inline app/views/admin/subjects/index.html.erb]: class=""

File: app/controllers/admin/subjects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  def new
+    @subject = Subject.new
+  end
+
+  def create
+    @subject = Subject.new subject_params
+    if @subject.save
+      redirect_to [:admin, @subject]
```
Comment (leminhtuan2015) [inline app/controllers/admin/subjects_controller.rb]: flash[:success] = t "Admin.create_success"

File: app/models/subject.rb [hunk 6]
```diff
@@ -1,7 +1,10 @@
 class Subject < ActiveRecord::Base
   belongs_to :user_subjects
-  has_many :tasks, dependent: :destroy
+  has_many :tasks, dependent: :destroy, inverse_of: :subject
+
+  accepts_nested_attributes_for :tasks, allow_destroy: :true
```
Comment (VoTaiTri) [inline app/models/subject.rb]: try 
```ruby 
reject_if
```
for nested_attributes

File: app/controllers/admin/subjects_controller.rb (hunk 1 above)
Comment (VoTaiTri) [inline app/controllers/admin/subjects_controller.rb]: try to user param
```ruby
:_destroy
```
for remove field in nested attributes

File: app/models/subject.rb [hunk 7]
```diff
@@ -1,7 +1,9 @@
 class Subject < ActiveRecord::Base
   belongs_to :user_subjects
-  has_many :tasks, dependent: :destroy
-  
-  validates :name, presence: true, lenght: {maximum: Settings.user.max}
+  has_many :tasks, dependent: :destroy, inverse_of: :subject
+
+  accepts_nested_attributes_for :tasks, allow_destroy: :true, reject_if: :all_blank
+  validates :name, presence: true, length: {maximum: Settings.user.maximum}
```
Comment (dieunb) [inline app/models/subject.rb]: slice to group code

File: app/models/task.rb [hunk 8]
```diff
@@ -1,7 +1,7 @@
 class Task < ActiveRecord::Base
-  belongs_to :subject
+  belongs_to :subject, inverse_of: :tasks
```
Comment (dieunb) [inline app/models/task.rb]: `, inverse_of: :tasks` no need

File: app/views/admin/subjects/_task_fields.html.erb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<div class="form-group">
+  <%= f.label :Task %>
+  <%= f.text_field :name, calss: "form-control" %>
+  <%= f.label :Description %>
+  <%= f.text_field :description, class: "form-control" %>
+  <div class="btn btn-info", id ="button">
```
Comment (dieunb) [inline app/views/admin/subjects/_task_fields.html.erb]: `id="button"`

**Pull request merged**